source/NVDAObjects/UIA/winConsoleUIA.py

```diff
@@ -297,12 +342,19 @@ def _get_windowThreadID(self):
 			threadID = super().windowThreadID
 		return threadID
 
+	def _get_is21H1Plus(self):
+		"Returns whether this is a newer version of Windows Console with an improved UIA implementation."
+		# microsoft/terminal#4495: In newer consoles,
+		# IUIAutomationTextRange::getVisibleRanges returns one visible range.
+		return self.UIATextPattern.GetVisibleRanges().length == 1
```

Comment:
Are you absolutely sure that pre21H1 will never return only one visible range?

Reply:
Only in a situation where the console displays *exactly* one line, which seems impossible.